### PR TITLE
Migrate CI builds to GitHub larger runners

### PR DESCRIPTION
**Pull request by str4d**

### CODE REVIEW

**Review by daira — DISMISSED** [review]
utACK

**Comment by str4d** [comment]
I've collected build times for this PR on the default GitHub runners (which have 2 cores), as well as their larger runners with 8 and 64 cores.

The groups of 12 bars are for the four build configurations (`base`, `with-libs`, `without-wallet`, `without-mining`) across the three platforms that the runners were building for (`arm64`, `ubuntu-22.04`, `windows`). The spike in the first of each group of four is expected (`base` is a full build, while the other three configurations are rebuilds with the altered config flags).

> The spike in the 2-core `window/base` build is lower on the "without caching" graph that the other spikes in its group because the CI run I used ([for the 5.7.0 merge to `master`](https://github.com/zcash/zcash/actions/runs/6353595174), to avoid needing to run a no-cache build on this PR with the default runners) found cached built dependencies for `windows` but not CCache files, while it found no caches for the other two platforms.

2023-11-14-6775-zcashd-build-times-uncached

For uncached builds we see around 45% decrease in initial build time and 55-60% decrease in rebuild time for a 4x increase in build cores, and a 25-30% decrease in IBT and 30-50% decrease in rebuild time moving from 8 to 64 cores (8x).

2023-11-14-6775-zcashd-build-times-cached

For cached builds we see a smaller improvement overall (which makes sense as the non-cached work is not as parallelizable): around 30% decrease in initial build time and 45% decrease in rebuild time for a 4x increase in build cores, and a 5% decrease in IBT and 13% decrease in rebuild time moving from 8 to 64 cores (8x).

Dependency updates and releases are the main sources of uncached builds; most other builds reliably hit the cache (as long as the relevant caches haven't been evicted). So I think for build purposes, the 8-core runners are the right sweet spot. We can determine whether more parallelization is helpful for running tests, when we migrate them from Tekton in subsequent PRs.

File: .github/workflows/audits.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: Audits
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches: master
```
Comment (yasserisa) [inline .github/workflows/audits.yml:6]: Using a regular expression might be advantageous if we wish to employ another branch, such as a "feature branch," and not restrict ourselves solely to a trunk-based model.

**Review by daira — APPROVED** [review]
utACK 96ea8e8cfb26dcd661aa9a4b4a274748248bc66a

**Review by yasserisa — COMMENTED** [review]
utACK

**Pull request merged**